socorro/unittest/processor/test_processor_2015.py

```diff
@@ -4,123 +4,153 @@
 from configman.dotdict import DotDict
 from mock import Mock, patch
 
-from socorro.lib.transform_rules import TransformRuleSystem
-from socorro.processor.processor_2015 import (
-    Processor2015,
-    rule_sets_from_string
-)
+from socorro.processor.processor_2015 import Processor2015
 from socorro.processor.general_transform_rules import (
     CPUInfoRule,
     OSInfoRule
 )
 from socorro.unittest.testbase import TestCase
 
 
-rule_set_01 = [
-    [
-        'ruleset01',
-        'tag0.tag1',
-        'socorro.lib.transform_rules.TransformRuleSystem',
-        'apply_all_rules',
-        'socorro.processor.general_transform_rules.CPUInfoRule, '
-        'socorro.processor.general_transform_rules.OSInfoRule '
-    ]
-]
-rule_set_01_str = ujson.dumps(rule_set_01)
-
-
 class TestProcessor2015(TestCase):
-    def test_rule_sets_from_string_1(self):
-        rule_set_config = rule_sets_from_string(rule_set_01_str)
-        rc = rule_set_config.get_required_config()
-        assert 'ruleset01' in rc
-        assert 'tag0.tag1' == rc.ruleset01.tag.default
-        expected = 'socorro.lib.transform_rules.TransformRuleSystem'
-        assert rc.ruleset01.rule_system_class.default == expected
-        assert 'apply_all_rules' == rc.ruleset01.action.default
-        expected = (
-            'socorro.processor.general_transform_rules.CPUInfoRule, '
-            'socorro.processor.general_transform_rules.OSInfoRule '
-        )
-        assert rc.ruleset01.rules_list.default == expected
 
     def test_Processor2015_init(self):
         cm = ConfigurationManager(
             definition_source=Processor2015.get_required_config(),
-            values_source_list=[{'rule_sets': rule_set_01_str}],
+            values_source_list=[],
         )
         config = cm.get_config()
         config.logger = Mock()
 
-        p = Processor2015(config)
+        Processor2015(config)
 
-        assert isinstance(p.rule_system, DotDict)
-        assert len(p.rule_system) == 1
-        assert 'ruleset01' in p.rule_system
-        assert isinstance(p.rule_system.ruleset01, TransformRuleSystem)
-        trs = p.rule_system.ruleset01
-        assert trs.act == trs.apply_all_rules
-        assert len(trs.rules) == 2
-        assert isinstance(trs.rules[0], CPUInfoRule)
-        assert isinstance(trs.rules[1], OSInfoRule)
-
-    def test_process_crash_no_rules(self):
+    def test_process_crash_existing_processed_crash(self):
         cm = ConfigurationManager(
             definition_source=Processor2015.get_required_config(),
-            values_source_list=[{'rule_sets': '[]'}],
+            values_source_list=[],
         )
         config = cm.get_config()
         config.logger = Mock()
+
         config.processor_name = 'dwight'
 
-        p = Processor2015(config)
+        p = Processor2015(config, rules=[
+            CPUInfoRule,
+            OSInfoRule,
+        ])
         raw_crash = DotDict()
+        raw_crash.uuid = "1"
```

Comment:
configman's `DotDict` lets you create things by passing in a dict. I tend to do this because I find it easier to read and uses less typing:

```
raw_crash = DotDict({
    'uuid': '1'
})
```

In this case it's not much easier to read, but I think it'll have a bigger effect in other cases.

Reply:
I was matching the lines below. I rewrote them too. Update incoming.